example.py

```diff
@@ -6,36 +7,52 @@
 from ape_tokens import tokens  # type: ignore[import]
 from taskiq import Context, TaskiqDepends, TaskiqState
 
-from silverback import AppState, CircuitBreaker, SilverbackApp
+from silverback import CircuitBreaker, SilverbackApp, StateSnapshot
 
 # Do this first to initialize your app
 app = SilverbackApp()
 
+# Cannot call `app.state` outside of an app function handler
+# app.state.something  # NOTE: raises AttributeError
+
 # NOTE: Don't do any networking until after initializing app
 USDC = tokens["USDC"]
 YFI = tokens["YFI"]
 
 
 @app.on_startup()
-def app_startup(startup_state: AppState):
-    # NOTE: This is called just as the app is put into "run" state,
-    #       and handled by the first available worker
-    # raise Exception  # NOTE: Any exception raised on startup aborts immediately
+def app_startup(startup_state: StateSnapshot):
+    # This is called just as the app is put into "run" state,
+    # and handled by the first available worker
+
+    # Any exception raised on startup aborts immediately:
+    # raise Exception  # NOTE: raises StartupFailure
+
+    # This is a great place to set `app.state` values
+    app.state.logs_processed = 0
+    # NOTE: Can put anything here, any python object works
+
     return {"block_number": startup_state.last_block_seen}
 
 
 # Can handle some resource initialization for each worker, like LLMs or database connections
 class MyDB:
     def execute(self, query: str):
-        pass
+        pass  # Handle query somehow...
 
 
 @app.on_worker_startup()
-def worker_startup(state: TaskiqState):  # NOTE: You need the type hint here
+def worker_startup(worker_state: TaskiqState):  # NOTE: You need the type hint to load worker state
+    # NOTE: Worker state is per-worker, not shared with other workers
     # NOTE: Can put anything here, any python object works
-    state.db = MyDB()
-    state.block_count = 0
-    # raise Exception  # NOTE: Any exception raised on worker startup aborts immediately
+    worker_state.db = MyDB()
+    worker_state.block_count = 0
```

Comment:
suggestion(non-blocking): to reduce confusion, maybe count something other than chain events which are shared between workers.

Reply:
I think we can just remove this line, it doesn't add anytihng important

also to your point about the over-use of the "state" word, I think I mostly eliminated it (e.g. refactor `AppState` to `StateSnapshot`), but this is still a problem (worker state) which I think we could fix by creating a higher-level mechanism around dependencies, which is really mostly what the worker startup is for